tools/unicode/c/genprops/layoutpropsbuilder.cpp

```diff
@@ -156,126 +239,62 @@ void LayoutPropsBuilder::checkMaxIntValue(UProperty prop, int32_t maxMax,
+    indexes[ULAYOUT_IX_VO_TRIE_TOP] = top += voLength;
 
-// Write one byte as one char in a Java String literal.
-// Java class file string literals use one byte per U+0001..U+007F,
-// and two bytes per U+0000 or U+0080..U+07FF.
-// This is reasonably compact if small byte values are more common, as usual.
-// Since 0 is very common but takes two bytes, we swap it with U+007A 'z'.
-int32_t appendByte(char *s, int32_t length, uint8_t b) {
-    if (b == 0) {
-        s[length++] = 'z';
-    } else if (b == 0x7a) {
-        s[length++] = '\\';
-        s[length++] = '0';
-    } else {
-        // Write all other bytes as octal escapes. (Java does not support \xhh.)
-        // We could make the source file smaller by writing ASCII characters
-        // directly where possible, but that would not make the class file any smaller,
-        // and we would have to be careful to still escape certain characters,
-        // and to escape digits after short octal escapes.
-        s[length++] = '\\';
-        if (b >= 0100) {
-            s[length++] = '0' + (b >> 6);
-        }
-        if (b >= 010) {
-            s[length++] = '0' + ((b >> 3) & 7);
-        }
-        s[length++] = '0' + (b & 7);
+    // Set reserved trie-top values to the top of the last trie
+    // so that they look empty until a later file format version
+    // uses one or more of these slots.
+    for (int32_t i = ULAYOUT_IX_RESERVED_TOP; i <= ULAYOUT_IX_TRIES_TOP; ++i) {
+        indexes[i] = top;
     }
-    return length;
-}
 
-void writeBytesAsJavaString(FILE *f, const uint8_t *bytes, int32_t length) {
-    // Quotes, line feed, etc., with up to 16 bytes per line, up to 4 bytes "\377" each.
-    char line[80];
-    int32_t lineLength = 0;
-    for (int32_t i = 0;;) {
-        if ((i & 0xf) == 0) {  // start of a line of 16 bytes
-            line[0] = '"';
-            lineLength = 1;
-        }
-        if (i < length) {
-            lineLength = appendByte(line, lineLength, bytes[i++]);
-        }
-        if (i == length) {  // end of the string
-            line[lineLength++] = '"';
-            line[lineLength++] = '\n';
-            line[lineLength++] = '\n';
-            line[lineLength++] = 0;
-            fputs(line, f);
-            break;
-        }
-        if ((i & 0xf) == 0) {  // end of a line of 16 bytes
-            line[lineLength++] = '"';
-            line[lineLength++] = ' ';
-            line[lineLength++] = '+';
-            line[lineLength++] = '\n';
-            line[lineLength++] = 0;
-            fputs(line, f);
-        }
-    }
-}
-
-static uint8_t trieBlock[100000];
-
-void writeUCPTrieAsJavaString(FILE *f, const UCPTrie *trie, UErrorCode &errorCode) {
-    int32_t length = ucptrie_toBinary(trie, trieBlock, UPRV_LENGTHOF(trieBlock), &errorCode);
-    writeBytesAsJavaString(f, trieBlock, length);
-}
-
-}  // namespace
+    indexes[ULAYOUT_IX_MAX_VALUES] =
+        ((getMaxIntValue(UCHAR_INDIC_POSITIONAL_CATEGORY)) << ULAYOUT_MAX_INPC_SHIFT) |
```

Comment:
I assume `getMaxIntValue` is guaranteed to return something no bigger than 0xff, right?

Reply:
LayoutPropsBuilder::build() calls checkMaxIntValue() for each property before we get here.